sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRequest.java

```diff
@@ -60,7 +61,22 @@ public HttpRequest(HttpMethod httpMethod, URL url, HttpHeaders headers, Flux<Byt
         this.httpMethod = httpMethod;
         this.url = url;
         this.headers = headers;
-        this.body = body;
+        setBody(BinaryDataHelper.createBinaryData(new FluxByteBufferContent(body)));
+    }
+
+    /**
+     * Create a new HttpRequest instance.
+     *
+     * @param httpMethod the HTTP request method
+     * @param url the target address to send the request to
+     * @param headers the HTTP headers to use with this request
+     * @param body the request content
+     */
+    public HttpRequest(HttpMethod httpMethod, URL url, HttpHeaders headers, BinaryData body) {
```

Comment:
Minor concern with this API is it could create ambiguous call sites when `Flux<ByteBuffer>` in the existing constructor was null. Case in point, the minimal constructor calling this constructor.

Reply:
added